atlasdb-api/src/main/java/com/palantir/atlasdb/util/MetricsManager.java

```diff
@@ -101,33 +105,25 @@ public void registerMetric(Class clazz, String metricName, Gauge gauge, Map<Stri
         taggedMetricRegistry.gauge(metricToAdd, gauge);
     }
 
-    public void registerMetric(Class clazz, String metricName, Metric metric) {
-        registerMetricWithFqn(MetricRegistry.name(clazz, metricName), metric);
-    }
-
-    public void registerGaugeForTable(Class clazz, String metricName, TableReference tableRef, Gauge gauge) {
-        Map<String, String> tag = getTableNameTagFor(tableRef);
-        registerMetric(clazz, metricName, gauge, tag);
-    }
-
     /**
-     * Add a new gauge metric of the given name.
+     * Add a new gauge metric of the given name or get the existing gauge if it is already registered.
      *
-     * If the metric already exists, this is a noop.
+     * If a non-gauge metric with the same name already exists, return the supplied gauge, but do not register it.
      */
-    public void registerIfNotExists(Class clazz, String metricName, Gauge gauge, Map<String, String> tag) {
+    public Gauge registerOrGet(Class clazz, String metricName, Gauge gauge, Map<String, String> tag) {
         MetricName metricToAdd = MetricName.builder()
                 .safeName(MetricRegistry.name(clazz, metricName))
                 .safeTags(tag)
                 .build();
 
         try {
-            taggedMetricRegistry.gauge(metricToAdd, gauge);
+            return taggedMetricRegistry.gauge(metricToAdd, gauge);
         } catch (IllegalArgumentException ex) {
             log.warn("Tried to add a gauge to a metric name {} that has non-gauge metrics associated with it."
                     + " This indicates a product bug.",
                     SafeArg.of("metricName", metricName),
                     ex);
+            return gauge;
```

Comment:
Why return if is not the one registered?

Reply:
The alternative is to throw a RTE, or return an Optional, but since the old behaviour was to only log at warn and ignore, this seemed to be more consistent. Not sure if we want to hard fail when we hit this